ocis-pkg/ldap/ldap.go

```diff
@@ -15,6 +16,20 @@ const (
 	caCheckSleep   = 2
 )
 
+var (
+	dnEscaper = strings.NewReplacer(
+		"\\", "\\\\",
+		",", "\\,",
+		"+", "\\+",
+		`"`, `\\"`,
+		"<", "\\<",
+		">", "\\>",
+		";", "\\;",
+		"=", "\\=",
+		"\000", "\\\000",
```

Comment:
If I am reading rfc4514 correctly the null byte as to be escaped as `\00` which would be:

```suggestion
		"\000", "\\00",
```

Only `' ', '"', '#', '+', ',', ';', '<', '=', '>', or '\'` can be escape by prepending a backslash/

Reply:
This is a tricky one but I argue that this is correct.

The left side is the "search" pattern so we are looking for the null byte which as a string is represented like `\000 (octal) or \x00 (hex)` both forms result in the null byte.

Now the right (replace) part is actually to things. First `\\` and second `\000` Together it results in `\00` but in string form it needs to be typed as it is above.

https://go.dev/play/p/IzDMsTunmaj

Reply:
Yeah, after researching a bit I found that you are right.
This also means that the python library I took inspiration from must be wrong :sweat_smile: : https://github.com/python-ldap/python-ldap/blob/44a593d1c55a007a43fcf30d2b027ac910ea1b96/Lib/ldap/dn.py#L31

Reply:
> This is a tricky one but I argue that this is correct.
> 
> The left side is the "search" pattern so we are looking for the null byte which as a string is represented like `\000 (octal) or \x00 (hex)` both forms result in the null byte.
> 
> Now the right (replace) part is actually to things. First `\\` and second `\000` Together it results in `\00` but in string form it needs to be typed as it is above.
> 
> https://go.dev/play/p/IzDMsTunmaj

Yeah, I see now why this doesn't make sense. :D